pyne/openmc_utils.py

```diff
@@ -156,3 +171,270 @@ def xml(self):
+
+    Parameters:
+    -----------
+    result : numpy array
+        This numpy array contains the flux data read from OpenMC state point
+        file. The shape of this numpy array is (num_ves*num_e_groups) or
+        (num_ves, ), with x changes the fastest.
+    dims : list
+        The dimension of x, y, z.
+
+    Returns:
+    --------
+    result_ : numpy array
+        This numpy array contains the flux data read from OpenMC state point
+        file. The shape of this numpy array is (num_ves*num_e_groups) or
+        (num_ves, ), with changes the fastest.
+    """
+
+    result_ = np.zeros_like(result)
+    x_dim, y_dim, z_dim = dims[0], dims[1], dims[2]
+    for i in range(x_dim):
+        for j in range(y_dim):
+            for k in range(z_dim):
+                idx = k * (x_dim * y_dim) + j * x_dim + i
+                idx_ = i * (y_dim * z_dim) + j * z_dim + k
+                result_[idx_] = result[idx]
+    return result_
+
+
+def create_meshtally(filename, tally_id, mesh_id=None, particle=None,
+                     tag_names=None, mesh_has_mats=False):
+    """
+    This function creates a MeshTally instance from OpenMC statepoint file.
+
+    Parameters:
+    -----------
+    filename : str
+        Filename of the OpenMC statepoint file. It ends with ".h5",
+        eg: "statepoint.10.h5".
+    tally_id : int
+        Tally number.
+    mesh_id : int
+        Mesh id of this tally used. Required if multiple meshes exist in the
+        OpenMC state point file.
+    particle : str
+        The particle type, 'neutron' or 'photon'.
+    tag_names : iterable, optional
+        Four strs that specify the tag names for the results, relative
+        errors, total results and relative errors of the total results.
+    mesh_has_mats: bool
+        If false, Meshtally objects will be created without PyNE material
+        objects.
+
+    Returns:
+    --------
+    m : MeshTally object
+        The MeshTally object created from OpenMC state point file with tally
+        number of tally_id.
+    """
+
+    m = MeshTally()
+    # assign tally_number
+    m.tally_number = tally_id
+    # assign particle
+    if particle != None:
+        m.particle = particle
+    # assign tag_names
+    if tag_names is None:
+        m.set_default_tag_names()
+    else:
+        m.tag_names = tag_names
+    # check tally_num exist
+    structured_coords = get_structured_coords_from_openmc_sp(
+        filename, tally_id=m.tally_number)
+
+    # parameters to create mesh
+    m.x_bounds = structured_coords[0]
+    m.y_bounds = structured_coords[1]
+    m.z_bounds = structured_coords[2]
+    m.dims = [len(m.x_bounds) - 1, len(m.y_bounds) - 1, len(m.z_bounds) - 1]
+    m.num_ves = (len(m.x_bounds)-1) * (len(m.y_bounds)-1) * (len(m.z_bounds)-1)
+    m.e_bounds = get_e_bounds_from_openmc_sp(filename, m.tally_number)
+    m.num_e_groups = len(m.e_bounds) - 1
+    mats = () if mesh_has_mats is True else None
```

Comment:
So what’s goin on here?

Reply:
It should report an error if `mesh_has_mats` is True, as it not yet supported. Updated in https://github.com/pyne/pyne/pull/1202/commits/d65a316f8b9d5f8782c9c280a76461bd3c70793d.